server/src/main/java/org/opensearch/search/startree/filter/provider/StarDateFieldMapper.java

```diff
@@ -0,0 +1,157 @@
+                timeZone,
+                forcedDateParser,
+                nowSupplier,
+                dateFieldType.resolution()
+            );
+            if (!rangeQueryBuilder.includeUpper()) {
+                --u;
+            }
+        }
+
+        // Find the matching interval - preferring the highest possible interval for query optimization
+        List<DateTimeUnitRounding> intervals = dateDimension.getSortedCalendarIntervals().reversed();
+        DateTimeUnitRounding matchingInterval = null;
+
+        for (DateTimeUnitRounding interval : intervals) {
+            // OpenSearch rounds up to the last millisecond in the rounding interval.
+            // So for example, closed-interval [l=2022-05-31T23:00:00.000, u=2022-05-31T23:59:59.999]
+            // (equivalent to half-open interval [l=2022-05-31T23:00:00.000, u+1=2022-06-01T00:00:00.000))
+            // can be resolved by star-tree 'hour' interval.
+            // To verify the above case, we check whether:
+            // 1/ l = 2022-05-31T23:00:00.000 is to the nearest 'hour' or not
+            // 2/ u+1 = 2022-06-01T00:00:00.000 is to the nearest 'hour' or not.
+
+            // Check if l is start of star-tree interval
+            boolean roundedLowMatches = l == Long.MIN_VALUE || interval.roundFloor(l) == l;
+
+            // Check if u+1 is the start of star-tree interval compared to u
+            boolean roundedHighMatches = u == Long.MAX_VALUE || interval.roundFloor(u + 1) == u + 1;
+
+            // If both bounds can be resolved by this star-tree interval, we have an exact match for this interval
+            if (roundedLowMatches && roundedHighMatches) {
+                matchingInterval = interval;
+                break; // Found the most granular matching interval
+            }
+        }
+
+        if (matchingInterval == null) {
+            return null; // No matching interval found, fall back to default implementation
+        }
+
+        // Construct the sub-dimension field name
+        subDimensionField = field + "_" + matchingInterval.shortName();
+
+        // l & u are inclusive interval boundaries here
+        return new RangeMatchDimFilter(field, l, u, true, true);
+    }
+
+    @Override
+    public Optional<String> getSubDimension() {
+        return Optional.ofNullable(subDimensionField);
+    }
+
+    @Override
+    public Optional<Long> getMatchingOrdinal(
+        String dimensionName,
+        Object value,
+        StarTreeValues starTreeValues,
+        DimensionFilter.MatchType matchType
+    ) {
+        // Since dates are stored as longs internally, we can treat them as numeric values
+        return Optional.of((Long) value);
+    }
+
+    @Override
+    public int compareValues(Object v1, Object v2) {
+        if (!(v1 instanceof Long) || !(v2 instanceof Long)) {
+            throw new IllegalArgumentException("Expected Long values for date comparison");
+        }
+        return Long.compare((Long) v1, (Long) v2);
```

Comment:
Why is no test covering this ?

Reply:
Is the case that UTs are not covering it but ITs are ?

Reply:
The abstract method was introduced recently in parent class. I had to implement it necessarily. Honestly, there is no utility of this yet - can just throw an UnsupportedException here as well.

Reply:
This was added as part of BooleanQuery support in StarTree which means there is no UT that contains a query on Date Field wrapped inside a BooleanQuery.

It is used for merging different sub-queries in a boolean query

Reply:
Added support + test cases for resolving boolean queries.